src/opnsense/service/templates/OPNsense/IDS/suricata.yaml

```diff
@@ -282,43 +260,18 @@ outputs:
             # for the body you need to set app-layer.protocols.smtp.mime.body-md5
             # to yes
             #md5: [body, subject]
-
-        #- dnp3
-        #- ftp
-        #- rdp
-        #- nfs
-        #- smb
-        #- tftp
-        #- ike
-        #- dcerpc
-        #- krb5
-        #- bittorrent-dht
-        #- snmp
-        #- rfb
-        #- sip
-        #- quic
-        #- dhcp:
-            #enabled: yes
+{%          elif opt == 'dhcp' and opt in eveLog_types_extended %}
             # When extended mode is on, all DHCP messages are logged
             # with full detail. When extended mode is off (the
             # default), just enough information to map a MAC address
             # to an IP address is logged.
-            #extended: no
-        - ssh
-        #- mqtt:
-            # passwords: yes           # enable output of passwords
-        #- http2
-        #- pgsql:
-            #enabled: no
-            # passwords: yes           # enable output of passwords. Disabled by default
-        #- stats:
-            #totals: yes       # stats for all threads merged together
-            #threads: no       # per thread stats
-            #deltas: no        # include delta values
-        # bi-directional flows
-        #- flow
-        # uni-directional flows
-        #- netflow
+            extended: yes
+{%          elif opt == 'mqtt' and opt in eveLog_types_extended %}
+            passwords: yes           # enable output of passwords
```

Comment:
this feels like a possible security concern...

Reply:
I agree it is a security concern but due to the fact it transits in plaintext over the wire, not due to its logging. A similar already existing risk is the logging of HTTP headers which might contain bearer tokens and/or cookies.

If you nonetheless object to this specific logging, I can remove it from the fields supporting extended logging.

Reply:
@0xThiebaut that's true, let me think about this, adding an extra warning in the help text might be enough and more generic anyway.